crypto.go

```diff
@@ -392,8 +417,11 @@ func PublicKeyFromStringEd25519(s string) (PublicKey, error) {
 	}, nil
 }
 
-// SLIP-10/BIP-32 Child Key derivation
-func _DeriveChildKey(parentKey []byte, chainCode []byte, index uint32) ([]byte, []byte) {
+func _DeriveEd25519ChildKey(parentKey []byte, chainCode []byte, index uint32) ([]byte, []byte) {
+	if IsHardenedIndex(index) {
+		panic("the index should not be pre-hardened")
```

Comment:
I know there are other places where we have `panic`, but we shouldn't use it. Returning error is the better approach.